velox/functions/sparksql/Comparisons.h

```diff
@@ -123,12 +123,33 @@ std::shared_ptr<exec::VectorFunction> makeGreaterThanOrEqual(
 
 inline std::vector<std::shared_ptr<exec::FunctionSignature>>
 comparisonSignatures() {
-  return {exec::FunctionSignatureBuilder()
-              .typeVariable("T")
-              .returnType("boolean")
-              .argumentType("T")
-              .argumentType("T")
-              .build()};
+  std::vector<std::shared_ptr<exec::FunctionSignature>> signatures;
+  signatures.push_back(exec::FunctionSignatureBuilder()
+                           .integerVariable("a_precision")
+                           .integerVariable("a_scale")
+                           .returnType("boolean")
+                           .argumentType("DECIMAL(a_precision, a_scale)")
+                           .argumentType("DECIMAL(a_precision, a_scale)")
+                           .build());
+  for (const auto& inputType :
+       {"tinyint",
+        "smallint",
+        "integer",
+        "bigint",
+        "hugeint",
+        "real",
+        "double",
+        "boolean",
+        "varchar",
+        "varbinary",
+        "timestamp"}) {
```

Comment:
Seems "date" could also be added.

Reply:
In Spark there are not "INTERVAL DAY TO SECOND" and "INTERVAL YEAR TO MONTH" types, right?

Reply:
OK, thank you.

Reply:
Updated, thanks.